modules/auxiliary/scanner/nntp/nntp_login.rb

```diff
@@ -0,0 +1,189 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Auxiliary::Report
+  include Msf::Auxiliary::AuthBrute
+  include Msf::Auxiliary::Scanner
+  include Msf::Exploit::Remote::Tcp
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'NNTP Login Utility',
+      'Description' => %q{
+        This module attempts to authenticate to NNTP services
+        which support the AUTHINFO authentication extension.
+
+        This module supports AUTHINFO USER/PASS authentication,
+        but does not support AUTHINFO GENERIC or AUTHINFO SASL
+        authentication methods.
+      },
+      'Author'      => 'Brendan Coles <bcoles[at]gmail.com>',
+      'License'     => MSF_LICENSE,
+      'References'  => [ [ 'CVE', '1999-0502' ], # Weak password
+                         [ 'URL', 'https://tools.ietf.org/html/rfc3977' ],
+                         [ 'URL', 'https://tools.ietf.org/html/rfc4642' ],
+                         [ 'URL', 'https://tools.ietf.org/html/rfc4643' ] ]))
+    register_options(
+      [
+        Opt::RPORT(119),
+        OptPath.new('USER_FILE', [ false, 'The file that contains a list of probable usernames.',
+          File.join(Msf::Config.install_root, 'data', 'wordlists', 'unix_users.txt') ]),
+        OptPath.new('PASS_FILE', [ false, 'The file that contains a list of probable passwords.',
+          File.join(Msf::Config.install_root, 'data', 'wordlists', 'unix_passwords.txt') ])
+      ])
+    deregister_options 'RHOST'
+  end
+
+  def run_host(ip)
+    begin
+      connect
+      return :abort unless nntp?
+      return :abort unless supports_authinfo?
+
+      report_service :host  => rhost,
+                     :port  => rport,
+                     :proto => 'tcp',
+                     :name  => 'nntp'
+      disconnect
```

Comment:
I know some protocols complain in the logs when a session is dropped without a  quit or similar message.  Unsure about NNTP.  Food for future thought if nothing else.